src/odemis/driver/technolution.py

```diff
@@ -144,14 +155,15 @@ def __init__(self, name, role, host, children, externalStorage, **kwargs):
         self.clockPeriod = model.FloatVA(1 / clockFrequencyData['frequency'], unit='s', readonly=True)
 
         # NOTE: Do not write real username/password here since this is published on github in plain text!
-        # example = ftp://username:password@127.0.0.1:5000/directory/sub-directory
+        # example = ftp://username:password@127.0.0.1:5000/directory/
+        # fixed url of the external storage configuration
         self.externalStorageURL = model.StringVA('ftp://%s:%s@%s/%s' %
                                                  (externalStorage["username"],
                                                   externalStorage["password"],
                                                   externalStorage["host"],
                                                   externalStorage["directory"]),
-                                                 setter=self._setURL)
-        self._setURL(self.externalStorageURL.value)  # Check and set the external storage URL to the ASM.
+                                                 setter=self._setURL, readonly=True)
+        self.externalStorageURL._set_value(self.externalStorageURL.value, force_write=True)
```

Comment:
I do think the comment on why setting a value which has been set in the line before is helping here.